DMCompiler/Compiler/DM/DMParserHelper.cs

```diff
@@ -21,6 +21,13 @@ public partial class DMParser {
             return level == ErrorLevel.Error;
         }
 
+        protected bool Error(WarningCode code, Location location, string message) {
+            ErrorLevel level = DMCompiler.CodeToLevel(code);
+            if (Emissions.Count < MAX_EMISSIONS_RECORDED)
+                Emissions.Add(new CompilerEmission(level, code, location, message));
+            return level == ErrorLevel.Error;
+        }
+
```

Comment:
Why use this over `DMCompiler.Emit()`?

The `Emissions` list is something we're trying to move away from.